kubernetes/resource_kubernetes_persistent_volume.go

```diff
@@ -245,6 +246,23 @@ func resourceKubernetesPersistentVolumeDelete(d *schema.ResourceData, meta inter
 		return err
 	}
 
+	err = resource.Retry(d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
+		out, err := conn.CoreV1().PersistentVolumes().Get(name, metav1.GetOptions{})
+		if err != nil {
+			if statusErr, ok := err.(*errors.StatusError); ok && statusErr.ErrStatus.Code == 404 {
```

Comment:
Minor nit:  the `== 404` is covered by the `IsNotFound(error)` checker function in `k8s.io/apimachinery/pkg/api/errors`.

Reply:
Maybe this could be implemented with an [Exists function](https://github.com/terraform-providers/terraform-provider-kubernetes/blob/78a8cac9911da8497f937f34db697dc2b7af1094/vendor/github.com/hashicorp/terraform/helper/schema/resource.go#L77):

```
	// Exists is a function that is called to check if a resource still
	// exists. If this returns false, then this will affect the diff
	// accordingly. If this function isn't set, it will not be called. It
	// is highly recommended to set it. The *ResourceData passed to Exists
	// should _not_ be modified.
```

That's what's done for DaemonSet and other resources: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/ee47b93c035892aad33ddfc7464caad3889d3654/kubernetes/resource_kubernetes_daemonset.go#L248

Not sure this is the intended role of such functions...

Reply:
Alternatively, I may move and reuse `waitForPersistenceVolumeDeleted()` from https://github.com/terraform-providers/terraform-provider-kubernetes/blob/a0e195447c3182b04785ebae4252204f152cdc65/kubernetes/resource_kubernetes_persistent_volume_test.go#L563

Reply:
Exists is indeed not a bad idea. Neither is re-using the waitFor from tests.

Reply:
Well, turns out there's already an `Exist` function so it does not help on deletion.
And `waitForPersistenceVolumeDeleted()` from tests is too tied to tests actually.

I've used `IsNotFound(err)`.